client/state/plugins/premium/actions.js

```diff
@@ -0,0 +1,268 @@
+	} );
+
+	if ( plugin.active ) {
+		dispatch( {
+			type: PLUGIN_SETUP_CONFIGURE,
+			siteId: site.ID,
+			slug: plugin.slug,
+		} );
+		configure( site, plugin, dispatch );
+		return;
+	}
+
+	getPluginHandler( site, plugin.slug ).install().then( ( data ) => {
+		dispatch( {
+			type: PLUGIN_SETUP_ACTIVATE,
+			siteId: site.ID,
+			slug: plugin.slug,
+		} );
+
+		data.key = plugin.key;
+		activate( site, data, dispatch );
+	} ).catch( ( error ) => {
+		if ( error.name === 'PluginAlreadyInstalledError' ) {
+			update( site, plugin, dispatch );
+		} else {
+			dispatch( {
+				type: PLUGIN_SETUP_ERROR,
+				siteId: site.ID,
+				slug: plugin.slug,
+				error,
+			} );
+			Dispatcher.handleServerAction( {
+				type: 'RECEIVE_INSTALLED_PLUGIN',
+				action: 'INSTALL_PLUGIN',
+				site: site,
+				plugin: plugin,
+				data: null,
+				error: error
+			} );
+		}
+	} );
+}
+
+function update( site, plugin, dispatch ) {
+	getPluginHandler( site, plugin.id ).updateVersion().then( ( data ) => {
+		dispatch( {
+			type: PLUGIN_SETUP_ACTIVATE,
+			siteId: site.ID,
+			slug: plugin.slug,
+		} );
+
+		data.key = plugin.key;
+		activate( site, data, dispatch );
+	} ).catch( ( error ) => {
+		dispatch( {
+			type: PLUGIN_SETUP_ERROR,
+			siteId: site.ID,
+			slug: plugin.slug,
+			error,
+		} );
+		Dispatcher.handleServerAction( {
+			type: 'RECEIVE_INSTALLED_PLUGIN',
+			action: 'INSTALL_PLUGIN',
+			site: site,
+			plugin: plugin,
+			data: null,
+			error: error
+		} );
+	} );
+}
+
+function activate( site, plugin, dispatch ) {
+	const success = ( data ) => {
+		dispatch( {
+			type: PLUGIN_SETUP_CONFIGURE,
+			siteId: site.ID,
+			slug: data.slug,
+		} );
+		Dispatcher.handleServerAction( {
+			type: 'RECEIVE_INSTALLED_PLUGIN',
+			action: 'INSTALL_PLUGIN',
+			site: site,
+			plugin: data,
+			data: data,
+			error: null
+		} );
+
+		autoupdate( site, data );
+		configure( site, plugin, dispatch );
+	};
+
+	getPluginHandler( site, plugin.id ).activate().then( success ).catch( ( error ) => {
+		if ( error.name === 'ActivationErrorError' ) {
+			// Technically it failed, but only because it's already active.
+			success( plugin );
+			return;
+		}
+		dispatch( {
+			type: PLUGIN_SETUP_ERROR,
+			siteId: site.ID,
+			slug: plugin.slug,
+			error,
+		} );
+		Dispatcher.handleServerAction( {
+			type: 'RECEIVE_INSTALLED_PLUGIN',
+			action: 'INSTALL_PLUGIN',
+			site: site,
+			plugin: plugin,
+			data: null,
+			error: error
+		} );
+	} );
+}
+
+function autoupdate( site, plugin ) {
+	getPluginHandler( site, plugin.id ).enableAutoupdate();
+}
+
+function configure( site, plugin, dispatch ) {
+	let option = false;
+	switch ( plugin.slug ) {
+		case 'vaultpress':
+			option = 'vaultpress_auto_register';
+			break;
+		case 'akismet':
+			option = 'wordpress_api_key';
+			break;
+		case 'polldaddy':
+			option = 'polldaddy_api_key';
+			break;
+	}
+	if ( ! option || ! plugin.key ) {
+		let optionError = new Error( 'We can\'t configure this plugin.' );
```

Comment:
is this error message going to be shown to the user? I can't find any place where it's being printed out, but maybe we plan to it sooner or later? In that case, we should translate it